src/categorical_algebra/CatElements.jl

```diff
@@ -44,13 +44,84 @@ function elements(X::StructACSet{S}) where S
 
   add_parts!(Y, :Hom, length(hom(S)), dom=dom_nums(S), cod=codom_nums(S), nameh=hom(S))
   map(enumerate(zip(hom(S), dom_nums(S), codom_nums(S)))) do (i, (f, ci, di))
-    c, d = obs[ci], obs[di]
-    nc = nparts(X, c)
+    c = obs[ci] #, d = obs[di]
+    # nc = nparts(X, c)
     add_parts!(Y, :Arr, nparts(X, c), src=els[ci], tgt=view(els[di], X[f]), πₐ=i)
   end
   return Y
 end
 
+
+"""    elements(f::ACSetTransformation)
+
+Apply category of elements functor to a morphism f: X->Y.
+This takes advantage of implementation details of elements of an object
```

Comment:
I'd say make it more explicit that this function depends on the `elements` constructor keeping elements of an object type together. That way if the `elements` function ever changes (breaking this assumption and the tests), it'll b easy to figure out what to change.